### PR TITLE
Tests with invalid chunks + CilentActor refactoring

### PR DESCRIPTION
**Pull request by ilblackdragon**
First step before implementing challenges is to add tests that showcase different cases of attack with invalid chunks.

Additionally split Client and ClientActor to make it easy to test without engaging in async behavior.

### CODE REVIEW

File: chain/chunks/src/lib.rs [hunk 1]
```diff
@@ -276,30 +298,26 @@ impl ShardsManager {
...
+    ) -> Result<(), Error> {
         let mut served = false;
         if let Some(chunk) = self.encoded_chunks.get(&request.chunk_hash) {
             if request.part_id as usize >= chunk.content.parts.len() {
-                return Err("Failed to process chunk part request: part_id is too big".into());
+                return Err(Error::InvalidChunkPartId);
             }
 
-            if let Some(part) = &chunk.content.parts[request.part_id as usize] {
+            if let Some(_) = &chunk.content.parts[request.part_id as usize] {
```
Comment (bowenwang1996) [inline chain/chunks/src/lib.rs]: `is_some`

File: chain/client/tests/challenges.rs [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    );
+    let prev_block_header = client.chain.get_header_by_height(0).unwrap();
+    let (block_with_invalid_chunk, _) = create_block_with_invalid_chunk(prev_block_header, "test2");
+    let (_, result) = client.process_block(block_with_invalid_chunk.clone(), Provenance::NONE);
+    // We have declined block with invalid chunk, but everyone who doesn't track this shard have accepted.
+    // At this point we should create a challenge and add it.
+    assert!(result.is_err());
+    assert_eq!(client.chain.head().unwrap().height, 0);
+
+    //    println!("{:?}", result);
```
Comment (bowenwang1996) [inline chain/client/tests/challenges.rs]: ?

File: chain/chunks/src/lib.rs [hunk 3]
```diff
@@ -30,13 +27,38 @@ use near_primitives::transaction::SignedTransaction;
...
+
+impl NetworkRecipient {
+    pub fn new(network_recipient: Recipient<NetworkRequests>) -> Self {
+        Self { network_recipient }
+    }
+}
+
+impl NetworkAdapter for NetworkRecipient {
+    fn send(&self, msg: NetworkRequests) {
+        let _ = self.network_recipient.do_send(msg);
```
[the remark this answers is not in the document]
Comment (ilblackdragon) [reply chain/chunks/src/lib.rs]: otherwise there is warning with result dropped.

File: chain/chunks/src/lib.rs [hunk 4]
```diff
@@ -746,6 +746,25 @@ impl ShardsManager {
         Ok(new_chunk)
     }
 
+    pub fn create_chunk_one_part(
```
Comment (ilblackdragon) [inline chain/chunks/src/lib.rs]: @Kouprin  can just use `encoded_chunk.create_chunk_one_part`

File: chain/client/tests/challenges.rs (hunk 2 above)
Comment (ilblackdragon) [reply chain/client/tests/challenges.rs]: need to write more tests out of that :)

**Pull request merged**